src/modules/SearchPage/components/MoveView/index.tsx

```diff
@@ -0,0 +1,194 @@
+import { useMoveContext, useSearchContext } from 'contexts';
+import { SortingSelector } from '../SortingSelector';
+import { ResultsSelector } from '../ResultsSelector';
+import { API, RESULTS_AMOUNT } from 'appConstants';
+import { MoveCard } from './components/MoveCard';
+import { Pagination } from '../Pagination';
+import { fetchAndMapMoves } from 'utils';
+import { Loader } from 'components';
+import { useMoveData } from 'hooks';
+import { MouseEvent } from 'react';
+import { MoveSorting } from 'types';
+import * as S from './styled';
+
+export const MoveView = () => {
+  const {
+    setSorting,
+    setBaseData,
+    setCurrentPage,
+    setResultsAmount,
+    setSearchResults,
+    setCurrentPageResults,
+    moveState: {
+      sorting,
+      baseData,
+      currentPage,
+      searchResults,
+      resultsAmount,
+      allDataResults,
+      currentPageResults,
+    },
+  } = useMoveContext();
+
+  const {
+    totalPageCount,
+    shouldFetchSearch,
+    sortById,
+    sortByPP,
+    sortByPower,
+    sortByAccuracy,
+    sortAplhabetically,
+  } = useMoveData();
+
+  const {
+    setIsLoading,
+    searchState: { isLoading },
+  } = useSearchContext();
+
+  const options = Object.values(MoveSorting);
+
```

Comment:
Try to avoid switch/case. Possible solution:
```
const sortObj = {
[MoveSorting.accuracy]: sortByAccuracy,
[MoveSorting.alphabetical]: sortAplhabetically,
[MoveSorting.power]: sortByPower,
[MoveSorting.pp]: sortByPP
};
const sort = (sorting: MoveSorting) => sortObj[sorting] ? sortObj[sorting]() : sortById();
```

Reply:
done